rkt/app_add.go

```diff
@@ -111,6 +112,15 @@ func runAppAdd(cmd *cobra.Command, args []string) (exit int) {
 	}
 	rktApps.Last().ImageID = *img
 
+	if rktApps.Last().Name == "" {
+		appName, err := common.ImageNameToAppName(types.ACIdentifier(args[1]))
```

Comment:
Not sure I like that app names include the version or result on weird things like `etcd-v2-0-0-linux-amd64-aci` since it's not consistent with anything we do.

You can add

```go
app.Name = appName.String()
```

here https://github.com/rkt/rkt/pull/3802/files#diff-71a7d6d46a80cd343d272145a7de0cf2L68

And get rid of the changes in this file (rkt/app_add.go) and things should work. So `https://github.com/coreos/etcd/releases/download/v2.0.0/etcd-v2.0.0-linux-amd64.aci` will have a name of `etcd` and `coreos.com/etcd:v2.0.10` too.

Wdyt?

Reply:
I don*t understand it. Why we need to do anything more in stage0? What change brings the explicit conversion to string? Will it magically cut versions from the name? I don't think so, if it doesn't work like that in our unit test - and the change you proposed is not going to change results of that unit test.

Reply:
My point is we're already calling `ImageNameToAppName` in stage0, but we're not using the name afterwards. And I don't want names like these:

```
75545b2a	etcd-v2-0-10			coreos.com/etcd:v2.0.10		exited	14 minutes ago	14 minutes ago	
		etcd-v2-0-0-linux-amd64-aci	coreos.com/etcd:v2.0.0							

```

Maybe it's easier to explain with code: https://github.com/kinvolk/rkt/commit/13f09448a6ecb3224b5c0ebfaa448ff3e84cc497

Reply:
To complete my comment, `ImageNameToAppName` should be applied to image names, not to image URLs like `coreos.com/etcd:v2.0.10` or `https://github.com/coreos/etcd/releases/download/v2.0.0/etcd-v2.0.0-linux-amd64.aci` and that's what we're doing in `rkt/app_add.go`